BrainPortal/app/controllers/tags_controller.rb

```diff
@@ -89,4 +88,18 @@ def destroy #:nodoc:
       format.xml  { head :ok }
     end
   end
+
+  private
+
+  # This method validates that the current user has access to the user and group being assigned to the created or updated tag
+  def validate_params #:nodoc:
+    user_id = params[:tag][:user_id]
+    group_id = params[:tag][:group_id]
+
+    params[:tag][:user_id]  = current_user.id           if !current_user.available_users.raw_first_column(:id).include?(user_id)
```

Comment:
I think there might be a bug here. Can you check? The params[:tag][:user_id] might be a string like "23" while the check in the if statement might generate a list of numbers, e.g. [5,9,23,99]. In that case, it means your code ALWAYS crush the values send in the form...